src/ws/test-channelresponse.c

```diff
@@ -833,21 +864,22 @@ test_resource_gzip_encoding (TestResourceCase *tc,
   g_assert_no_error (error);
 
   bytes = g_memory_output_stream_steal_as_bytes (tc->output);
-  cockpit_assert_bytes_eq (bytes,
-                           "HTTP/1.1 200 OK\r\n"
-                           "X-DNS-Prefetch-Control: off\r\nReferrer-Policy: no-referrer\r\n"
-                           "Content-Encoding: gzip\r\n"
-                           "Cache-Control: no-cache, no-store\r\n"
-                           "Access-Control-Allow-Origin: http://localhost\r\n"
-                           "Content-Type: text/plain\r\n"
-                           "Transfer-Encoding: chunked\r\n"
-                           "Vary: Cookie\r\n"
-                           "\r\n"
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "HTTP/1.1 200 OK\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*X-DNS-Prefetch-Control: off\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Referrer-Policy: no-referrer\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Content-Encoding: gzip\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Content-Type: text/plain\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Cache-Control: no-cache, no-store\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Access-Control-Allow-Origin: http://localhost\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Transfer-Encoding: chunked\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL), "*Vary: Cookie\r\n*");
+  cockpit_assert_strmatch (g_bytes_get_data (bytes, NULL),
```

Comment:
Thanks for doing this work. But somehow this stuff needs to make sure no additional headers are included.

Reply:
Asserting the number of headers (a.k.a the number of `\n`) should do the trick?

Reply:
This is very much a non ideal way yea. 

For `test-packages` I'm actually redoing it so it pops the headers first, compares those as json so the order doesn't matter, and then compares the content. That should make it a lot better.

For `test-channelresponse` the number of `\n`'s might work yea.